### PR TITLE
fix port name in gateway deployment

### PR DESCRIPTION
**Pull request by waynz0r**

### CODE REVIEW

File: pkg/resources/gateways/deployment.go [hunk 1]
```diff
@@ -156,7 +156,7 @@ func (r *Reconciler) ports(gw string) []apiv1.ContainerPort {
 			{ContainerPort: 443, Protocol: apiv1.ProtocolTCP, Name: "https"},
 			{ContainerPort: 31400, Protocol: apiv1.ProtocolTCP, Name: "tcp"},
 			{ContainerPort: 15029, Protocol: apiv1.ProtocolTCP, Name: "https-kiali"},
-			{ContainerPort: 15030, Protocol: apiv1.ProtocolTCP, Name: "https-prometheus"},
+			{ContainerPort: 15030, Protocol: apiv1.ProtocolTCP, Name: "https-prom"},
```
Comment (Laci21) [inline pkg/resources/gateways/deployment.go]: Why did you change this name? Shouldn't you change it [here](https://github.com/banzaicloud/istio-operator/blob/istio11-fix-gateway-deployment/pkg/resources/gateways/service.go#L48) as well then?

Comment (waynz0r) [reply pkg/resources/gateways/deployment.go]: I've got the following error, `could not reconcile istio: failed to reconcile resource: updating resource failed: Deployment.apps \"istio-ingressgateway\" is invalid: spec.template.spec.containers[0].ports[4].name: Invalid value: \"https-prometheus\": must be no more than 15 characters`

Comment (Laci21) [reply pkg/resources/gateways/deployment.go]: Ah, I see. Thanks for the commit!

Comment (waynz0r) [reply pkg/resources/gateways/deployment.go]: For some reason that limit is different at the service definition, but i changed the name there as well.
